src/libAtomVM/term.h

```diff
@@ -242,24 +242,24 @@ static inline int term_is_boxed(term t)
 /**
  * @brief Checks if a term is a movable boxed value
  *
- * @details Returns 1 if a term is a boxed value that can be safely copied with memcpy.
+ * @details Returns \c true if a term is a boxed value that can be safely copied with memcpy.
  * @param t the term that will checked.
- * @return 1 if check succeeds, 0 otherwise.
+ * @return \c true if check succeeds, \c false otherwise.
  */
-static inline int term_is_movable_boxed(term t)
+static inline bool term_is_movable_boxed(term t)
 {
     /* boxed: 10 */
     if ((t & 0x3) == 0x2) {
         const term *boxed_value = term_to_const_term_ptr(t);
         switch (boxed_value[0] & TERM_BOXED_TAG_MASK) {
```

Comment:
Would it make sense to simplify this switch and replace it with a boolean expression?

Reply:
I think I did it this way, so I can cover more kinds of movable boxed, however I didn't implement that.
So maybe we can improve performances around.